tools/torch/data.lua

```diff
@@ -41,33 +41,205 @@ local function all_keys(cursor_,key_,op_)
         end)
 end
 
-local PreProcess = function(y, meanTensor, mirror, crop, train, cropY, cropX, croplen)
+-- HSV Augmentation
+-- Parameters:
+-- @param im (tensor): input image
+-- @param augHSV (table): standard deviations under {H,S,V} keys with float values.
+local augmentHSV = function(im_rgb, augHSV)
+    -- Fair augHSV standard deviation values are {H=0.02,S=0.04,V=0.08}
+    local im_hsv = image.rgb2hsv(im_rgb)
+    if augHSV.H >0.001 then
```

Comment:
Hehe there's a reason. Because any value below 0.001 will do so little it will make no sense at all. So it it might save some computation time here and there, especially with very little H-augmentation and a standard deviation like 0.01. For readability sake I could also take it out.